quicktill/pdrivers.py

```diff
@@ -701,6 +717,50 @@ def process_canvas(self, canvas, f):
                 f.write(escpos.ep_fullcut)
             f.flush()
 
+    def _image(self, data, width, height, f):
+        # Print a PBM bit-image
+        if width > self.dpl:
+            # Image too wide for paper
+            return
+
+        # Calculate padding required to center the image
+        padding = (self.dpl - width) // 2
+        padchars = [False] * padding
+
+        # Partition the bitmap into lines
+        lines = []
+        for line in range(height):
+            start = line * (width // 8)
+            end = start + (width // 8)
+            line = padchars.copy()
+            for byte in data[start:end]:
+                for bit in f"{int(byte):08b}":
+                    line.append(bool(int(bit)))
+            lines.append(line)
+
+        # Compact up to twenty-four bit-lines into each row
+        rows = []
+        for chunk in range(height // 24):
```

Comment:
A more Pythonic way to do this would be to iterate over a generator that returns (up-to) 24 row chunks.

Eg. 
```
for chunk in itertools.batched(lines, 24):
    ...
    for column in zip(*chunk):
        ...
```

Reply:
Thanks!  Much better :+1:

Reply:
What's the Python version support policy for the application?  I see Py3.8+ mentioned in `setup.cfg`; `itertools.batched` seems to be from Py3.12+ (still possible that a cleanup similar to this is possible using more backwards-compatible functions, though).

Reply:
Oh, good point. Yes, `itertools.batched` is still too new to use. You could include a generator that does something suitable, and which also pads to a multiple of 24 rows?

Reply:
(And to answer your question directly: I'm still officially supporting Python 3.8 [it's in Ubuntu 20.04 which I still use in a couple of places], but I expect I'll drop that for the next major release and require a minimum of Python 3.10.)

Reply:
Aha: [`more-itertools`](https://pypi.org/project/more-itertools/) provides `batched`, and in fact may even be the origin of the Py3.12 implementation (although I'm not certain of that).

It's Py3.8+ compatible, and using a similar implementation would make it easier to drop that dependency when the baseline here becomes Py3.12+ at a later date.

Reply:
Unfortunately the version of `more-itertools` in Ubuntu 22.04 (which the tills run on in my pub company) doesn't include `batched`. I think you're going to have to include the text of the function in this project. (It's very small.)

Reply:
D'oh, that's frustrating.  I noticed this comment _after_ adding the dependency to the GitHub Workflows definition.. clearing that out in a moment.